From 8c88be192b567ec30a75d814c61cee53faaad155 Mon Sep 17 00:00:00 2001
From: Victor Rodriguez <victor.rodriguez.bahena@intel.com>
Date: Mon, 11 Dec 2017 21:23:17 +0000
Subject: [PATCH] smpboot: reuse timer calibration saving boot time

x86 SMP booting functions ask for timer calibration. If the Time Stamp
Counter (TSC) is a known-constant there is not point recalibrating

This patch saves ~200ms+ of boot time.

Patch adapted to fix broken build in 4.9.68 linux-lts/linux-hyperv-lts

Author:    Arjan van de Ven <arjan@linux.intel.com>

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Signed-off-by: Victor Rodriguez <victor.rodriguez.bahena@intel.com>
---
 arch/x86/kernel/tsc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/x86/kernel/tsc.c b/arch/x86/kernel/tsc.c
index 44bf5cf417d3..e0b7e269b7b6 100644
--- a/arch/x86/kernel/tsc.c
+++ b/arch/x86/kernel/tsc.c
@@ -1388,6 +1388,9 @@ unsigned long calibrate_delay_is_known(void)
 	if (tsc_disabled || !constant_tsc || !mask)
 		return 0;
 
+    if (cpu != 0)
+        return cpu_data(0).loops_per_jiffy;
+
 	sibling = cpumask_any_but(mask, cpu);
 	if (sibling < nr_cpu_ids)
 		return cpu_data(sibling).loops_per_jiffy;
-- 
2.15.1

